std/process.d

```diff
@@ -406,9 +414,42 @@ private Pid spawnProcessImpl(in char[][] args,
     auto stdoutFD = getFD(stdout);
     auto stderrFD = getFD(stderr);
 
+    int[2] forkPipe;
+    if (core.sys.posix.unistd.pipe(forkPipe) == 0)
+    {
+        setCLOEXEC(forkPipe[1], true);
+    }
+    else
+    {
+        throw ProcessException.newFromErrno("Could not create pipe to check startup of child");
+    }
+    scope(exit) close(forkPipe[0]);
+
+    static void abortOnError(int forkPipeOut, InternalError errorType, int error) nothrow
+    {
+        core.sys.posix.unistd.write(forkPipeOut, &errorType, errorType.sizeof);
+        core.sys.posix.unistd.write(forkPipeOut, &error, error.sizeof);
+        close(forkPipeOut);
+        core.sys.posix.unistd._exit(1);
+        assert(0);
+    }
+
+    static void ignorePipeErrors() nothrow
+    {
+        import core.sys.posix.signal;
+        import core.stdc.string : memset;
+        sigaction_t ignoreAction;
+        memset(&ignoreAction, 0, sigaction_t.sizeof);
+        ignoreAction.sa_handler = SIG_IGN;
+        sigaction(SIGPIPE, &ignoreAction, null);
```

Comment:
> Should we care about the case when parent process dies before forked child proceeds to execve?

I can't think of a reasonable situation where it would make a difference, so I don't think so. If the parent process is gone, the child process will be disowned, in which case its exit code doesn't matter. Furthermore, unless the termination of the parent process is somehow linked with the `spawnProcess` call, it would be a random event, in which case it wouldn't matter whether the parent process was terminated before the `fork` call, or after it but before it received the child status. And if we're going to worry about either process being randomly terminated, then that also raises the question of how to handle the child process being killed before it communicates its status... not something we should worry about, I think.